src/Conversion/ONNXToMhlo/Math/Clip.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===----------------- Clip.cpp - Lowering Clip Op ------------------------===//
+//
+// Copyright 2022
+//
+// =============================================================================
+//
+// This file lowers the ONNX Clip Operator to Mhlo dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToMhlo/DialectBuilder.hpp"
+#include "src/Conversion/ONNXToMhlo/ONNXToMhloCommon.hpp"
+#include "src/Dialect/ONNX/ONNXOps/NewShapeHelper.hpp"
+#include "src/Support/TypeUtilities.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+//===----------------------------------------------------------------------===//
+// Scalar unary ops for lowering ONNXClipOp
+//===----------------------------------------------------------------------===//
+
+struct ONNXClipOpLoweringToMhlo : public ConversionPattern {
+  ONNXClipOpLoweringToMhlo(MLIRContext *ctx)
+      : ConversionPattern(ONNXClipOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    Location loc = op->getLoc();
+
+    ONNXClipOpAdaptor operandAdaptor(operands);
+    IndexExprBuilderForMhlo createIE(rewriter, loc);
+    NewONNXClipOpShapeHelper shapeHelper(op, operands, &createIE);
+    auto shapeComputed = shapeHelper.computeShape();
+    assert(succeeded(shapeComputed) && "Could not compute output shape");
+
+    Value input = operandAdaptor.input();
+    Value min = operandAdaptor.min();
+    Value max = operandAdaptor.max();
+
+    Type outputType = *op->result_type_begin();
+    assert(isRankedShapedType(outputType) && "Expected Ranked ShapedType");
+    ShapedType outputShapedType = outputType.cast<ShapedType>();
+    Type elemType = outputShapedType.getElementType();
+
+    MathBuilder createMath(rewriter, loc);
+    if (min.getType().isa<NoneType>()) {
```

Comment:
FYI: #1965 is recommending not tu use `isa<NoneType>` anymore because it does not cover all the cases in ONNX. See PR for latest recommendation, in case your PR comes in after.